operator/controllers/keptnworkloadinstance/controller.go

```diff
@@ -275,35 +274,45 @@ func (r *KeptnWorkloadInstanceReconciler) getAppVersion(ctx context.Context, app
 func (r *KeptnWorkloadInstanceReconciler) getAppVersionForWorkloadInstance(ctx context.Context, wli *klcv1alpha1.KeptnWorkloadInstance) (bool, klcv1alpha1.KeptnAppVersion, error) {
 	apps := &klcv1alpha1.KeptnAppVersionList{}
 
+	// TODO add label selector for looking up by name?
 	if err := r.Client.List(ctx, apps, client.InNamespace(wli.Namespace)); err != nil {
 		return false, klcv1alpha1.KeptnAppVersion{}, err
 	}
+
+	workloadFound, latestVersion, err := getLatestAppVersion(apps, wli)
+	if err != nil {
+		r.Log.Error(err, "could not look  up KeptnAppVersion for WorkloadInstance")
+		return false, latestVersion, err
+	}
+
+	if latestVersion.Spec.Version == "" || !workloadFound {
+		return false, klcv1alpha1.KeptnAppVersion{}, nil
+	}
+	return true, latestVersion, nil
+}
+
+func getLatestAppVersion(apps *klcv1alpha1.KeptnAppVersionList, wli *klcv1alpha1.KeptnWorkloadInstance) (bool, klcv1alpha1.KeptnAppVersion, error) {
 	latestVersion := klcv1alpha1.KeptnAppVersion{}
+	// ignore the potential error since this can not return an error with 0.0.0
+	oldVersion, _ := version.NewVersion("0.0.0")
+
+	workloadFound := false
 	for _, app := range apps.Items {
 		if app.Spec.AppName == wli.Spec.AppName {
-
 			for _, appWorkload := range app.Spec.Workloads {
-				if !reflect.DeepEqual(latestVersion, app) {
-					latestVersion = app
-				} else if appWorkload.Version == wli.Spec.Version && fmt.Sprintf("%s-%s", app.Spec.AppName, appWorkload.Name) == wli.Spec.WorkloadName {
-					oldVersion, err := version.NewVersion(app.Spec.Version)
-					if err != nil {
-						r.Log.Error(err, "could not parse version")
-					}
-					newVersion, err := version.NewVersion(latestVersion.Spec.Version)
+				if appWorkload.Version == wli.Spec.Version && fmt.Sprintf("%s-%s", app.Spec.AppName, appWorkload.Name) == wli.Spec.WorkloadName {
```

Comment:
can we export `fmt.Sprintf("%s-%s", app.Spec.AppName, appWorkload.Name)` to a separate function?

the usage will be something like :

`app.GetWorkloadName(appWorkload.Name)`